File: libs/language/src/lib/mapping/__tests__/schema/y1.ts
import z from 'zod';
import { x11Schema } from './x11';
import { x12Schema } from './x12';
import { ySchema } from './y';
import { withTypeBlueId } from '../../../../schema/annotations';

export const y1Schema = withTypeBlueId('Y1-BlueId')(
  ySchema.extend({
    x11Field: x11Schema.optional(),
    x12Field: x12Schema.optional(),
    x11ListField: z.array(x11Schema).optional(),
  })
);

File: references/java-sdk/src/test/java/blue/language/mapping/model/Y1.java
package blue.language.mapping.model;

import blue.language.model.TypeBlueId;

import java.util.List;

@TypeBlueId("Y1-BlueId")
public class Y1 extends Y {
    public X11 x11Field;
    public X12 x12Field;
    public List<X11> x11ListField;
}